containers/keys/shared/flags.ts

```diff
@@ -0,0 +1,20 @@
+import { KEY_FLAG } from 'proton-shared/lib/constants';
+import { FlagAction } from './interface';
+
+const { SIGNED, ENCRYPTED_AND_SIGNED, CLEAR_TEXT } = KEY_FLAG;
+
+export const getNewKeyFlags = (action: FlagAction) => {
```

Comment:
Written this way you don't need to throw, and ts understands correctly, because every value from one enum maps to everything  in another.

```
export const getNewKeyFlags = (action: FlagAction) => {
    return {
        [FlagAction.MARK_OBSOLETE]: SIGNED,
        [FlagAction.MARK_NOT_OBSOLETE]: ENCRYPTED_AND_SIGNED,
        [FlagAction.MARK_COMPROMISED]: CLEAR_TEXT,
        [FlagAction.MARK_NOT_COMPROMISED]: SIGNED,
    }[action]
};
```